uwblueprint/ccbc
# Sebastian/setup-and-creator-overview-section

## Notion ticket link

[Booker: Setup + Creator Overview Section](https://www.notion.so/uwblueprintexecs/16fb020b93f641c8b384f76a1b3c769e?v=0f0df1f1b7074d2cb19b7eb7652d7d04&p=71b3cec01f074c2da7e22e9f0544a371&pm=c)



## Implementation description
* Created route with react Router with the url ` "/creators/:id"`
* Created a function in the CreatorAPIClient.ts called `getCreatorByID` to access each creator from the ID pulled from the url params
* Under` backend/typescript/rest/creatorRoutes.ts`, there was a new endpoint that needed to added because the existing one did not work properly.
* `CreatorProfile `  and  `CreatorOverview`  and component was created modelling the figma prototypes



## Steps to test
1. Ensure the table 'creator' in the postgres database is populated with creators
1. Run docker-compose up
2. Go to localhost:3000/creators/1
3. Test to see if the correct data and layout are present



## What should reviewers focus on?
* Double check the added code in `creatorRoutes `is up to standards
* Contact button has no function yet
* Creators do not have images in their data model. Where will we be getting these from?
* Creator data such as Craft and Genre are currently strings, however, in the [figma](https://www.figma.com/file/IGLWQQbCrEkszRgTKdkucK/Creator-Booking---Booker-Experience?node-id=972%3A12627&t=ck3m5p80UJUjpzvI-0) they seem to be listed items. Rendering will need to be changed to loop over these items.
* Should error handling for empty data fields be implemented or will empty values work?


## Checklist
